File: c/2/sumefibs.c
#include "sumefibs.h"

int sum_efibs(int limit) {
    int sum = 0;
    int a = 1;
    int b = 1;
    int c = a + b;

    while (c < limit) {
        sum = sum + c;
        a = b + c;
        b = c + a;
        c = a + b;
    }

    return sum;
}

File: haskell/2/sumefibs.hs
efib :: Int -> Integer
efib 0 = error "There's no 0th even fibonacci number."
efib 1 = 2
efib 2 = 8
efib x = 4 * efib (x-1) + efib (x-2)

sumEvenFibs x = sum (takeWhile (<x) (map efib [1,2..]))
